opentelemetry-sdk/src/opentelemetry/sdk/util/__init__.py

```diff
@@ -45,6 +46,10 @@ def get_dict_as_key(labels):
     )
 
 
+def make_immutable_dict(attributes):
```

Comment:
I would just put this logic directly in the `trace` file. Otherwise you should probably replace all other uses of immutability with this function.